### PR TITLE
fix(sanitizers): failing json test

### PR DESCRIPTION
**Pull request by kostasrim**
* fix sanitizers for json MGET tests

### CODE REVIEW

**Comment by kostasrim** [comment]
actions https://github.com/dragonflydb/dragonfly/actions/runs/9513408447

File: src/core/json/lexer_impl.cc [hunk 1]
```diff
@@ -6,8 +6,6 @@
 
 #include <absl/strings/str_cat.h>
 
-using namespace std;
```
Comment (kostasrim) [inline src/core/json/lexer_impl.cc:9]: DO NOT USE `using namespace std` in source files. 

It did not pull `std::string` on the function below.... :facepalm:

Comment (kostasrim) [reply src/core/json/lexer_impl.cc:9]: Basically stop triggering `ADL` unless you are writing customization points.

Comment (kostasrim) [reply src/core/json/lexer_impl.cc:9]: I was `wrong`, it's not `ADL`. It's a false positive and I stepped through it and it's an `std::string`. 

Still I don't like pulling namespaces in source files :stuck_out_tongue:

File: src/server/json_family_test.cc [hunk 2]
```diff
@@ -952,8 +952,10 @@ TEST_F(JsonFamilyTest, MGet) {
   resp = Run({"JSON.SET", "json2", ".", json[1]});
   ASSERT_THAT(resp, "OK");
 
+#ifndef SANITIZERS
```
Comment (romange) [inline src/server/json_family_test.cc:955]: is it a good solution? Do you know why it failed?

**Comment by romange** [comment]
I suggest to disable this particular feature altogether:
https://github.com/google/sanitizers/wiki/AddressSanitizerUseAfterReturn
i.e. `ASAN_OPTIONS=detect_stack_use_after_return=0`

File: src/server/json_family_test.cc (hunk 2 above)
Comment (kostasrim) [reply src/server/json_family_test.cc:955]: I am fairly certain that it's a false positive. I thought it was ADL and a funky conversion somewhere but I was wrong on that. I will spent a little bit time on this and ping back just to be on the safe side

**Comment by kostasrim** [comment]
> I suggest to disable this particular feature altogether: https://github.com/google/sanitizers/wiki/AddressSanitizerUseAfterReturn i.e. `ASAN_OPTIONS=detect_stack_use_after_return=0`

I understand that this is flaky because of the false positives but it could also be useful sometimes and I don't mind spending a few minutes to investigate when this happens. I don't have a strong opinion on this so I am happy to disable this.

**Comment by romange** [comment]
I prefer disabling this specific check since we know that asan do not work well with fibers - the noise is not worth it  to micro optimize those tests

File: src/server/json_family_test.cc (hunk 2 above)
Comment (kostasrim) [reply src/server/json_family_test.cc]: It is a false positive, I stepped it through in debug build and everything is fine in the flow. I switched off the check

**Comment by romange** [comment]
LGTM
When I have time I will try to overlay the info in this thread: 
https://github.com/google/sanitizers/issues/1760

and see if it can fix the problem

**Review by romange — APPROVED** [review]

**Pull request merged**